src/Umbraco.Web/PublishedElementExtensions.cs

```diff
@@ -165,8 +165,7 @@ public static IndexedArrayItem<TContent>[] ToIndexedArray<TContent>(this IEnumer
         public static IEnumerable<T> OfTypes<T>(this IEnumerable<T> contents, params string[] types)
             where T : IPublishedElement
         {
-            types = types.Select(x => x.ToLowerInvariant()).ToArray();
-            return contents.Where(x => types.Contains(x.ContentType.Alias.ToLowerInvariant()));
+            return contents.Where(x => types.InvariantContains(x.ContentType.Alias));
```

Comment:
Should we validate and short-circuit if `types` is `null` or an empty array?

```suggestion
            if (types == null || types.Length == 0) return Enumerable.Empty<T>();
            
            return contents.Where(x => types.InvariantContains(x.ContentType.Alias));
```